src/Microsoft.Azure.WebJobs.Host/Bindings/SysBindingData.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Microsoft.Azure.WebJobs.Host.Bindings.Path;
+
+namespace Microsoft.Azure.WebJobs.Host.Bindings
+{
+    /// <summary>
+    /// Built in object for all binding expressions. 
+    /// </summary>
+    /// <remarks>
+    /// It's expected this class is created and added to the binding data. 
+    /// </remarks>
+    internal class SysBindingData
+    {
+        // The name for this binding in the binding expressions. 
+        public const string Name = "sys";
+
+        // A name that can't be overwritten by the user that we cna place in the binding data dictionary
+        // and use to unambiguously retrieve this later. 
+        private const string InternalName = "$sys";
+
+        private static readonly IReadOnlyDictionary<string, Type> DefaultSysContract = new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase)
+        {
+            { Name, typeof(SysBindingData) }
+        };
+
+        /// <summary>
+        /// The method name that the binding lives in. 
+        /// The method name can be override by the <see cref="FunctionNameAttribute"/> 
+        /// </summary>
+        public string MethodName { get; set; }
```

Comment:
I'd love to do that, but it's going to be tricky because we still need to support the existing "{datetime:format}" expression and "{rand-guid}x{rand-guid}" semantics.  I didn't see how to do that without keeping the existing BuiltinResolver  classes. 


---
In reply to: [117077937](https://github.com/Azure/azure-webjobs-sdk/pull/1148#discussion_r117077937) [](ancestors = 117077937)